exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/impl/NetworkFunctions.java

```diff
@@ -0,0 +1,566 @@
+      buffer.setBytes(0, outputValue.getBytes());
+
+    }
+  }
+
+  /**
+   * This function decodes URL strings.
+   */
+  @FunctionTemplate(name = "url_decode", scope = FunctionTemplate.FunctionScope.SIMPLE, nulls = FunctionTemplate.NullHandling.NULL_IF_NULL)
+  public static class urldecodeFunction implements DrillSimpleFunc {
+
+    @Param
+    VarCharHolder inputString;
+
+    @Output
+    VarCharHolder outputString;
+
+    @Inject
+    DrillBuf buffer;
+
+    public void setup() {
+    }
+
+    public void eval() {
+
+      String url = org.apache.drill.exec.expr.fn.impl.StringFunctionHelpers.toStringFromUTF8(inputString.start, inputString.end, inputString.buffer);
+
+      String outputValue = "";
+      try {
+        outputValue = java.net.URLDecoder.decode(url, "UTF-8");
+      } catch (Exception e) {
+
+      }
+      outputString.buffer = buffer;
+      outputString.start = 0;
+      outputString.end = outputValue.getBytes().length;
+      buffer.setBytes(0, outputValue.getBytes());
+
+    }
+  }
+
+
+  /**
+   * This function converts a BigInt IPv4 into dotted decimal notation.  The opposite of inet_aton.
+   */
+
+  @FunctionTemplate(name = "inet_ntoa", scope = FunctionTemplate.FunctionScope.SIMPLE, nulls = FunctionTemplate.NullHandling.NULL_IF_NULL)
+  public static class InetNtoaFunction implements DrillSimpleFunc {
+
+    @Param
+    BigIntHolder in;
+
+    @Output
+    VarCharHolder out;
+
+    @Inject
+    DrillBuf buffer;
+
+
+    public void setup() {
+    }
+
+
+    public void eval() {
+      StringBuilder result = new StringBuilder(15);
+
+      long inputInt = in.value;
+
+      for (int i = 0; i < 4; i++) {
+
+        result.insert(0, Long.toString(inputInt & 0xff));
+
+        if (i < 3) {
+          result.insert(0, '.');
+        }
+
+        inputInt = inputInt >> 8;
+      }
+
+      String outputValue = result.toString();
+
+      out.buffer = buffer;
+      out.start = 0;
+      out.end = outputValue.getBytes().length;
+      buffer.setBytes(0, outputValue.getBytes());
+    }
+
+
+  }
+
+  /**
+   * This function returns true if a given IPv4 address is private, false if not.
+   */
+
+  @FunctionTemplate(name = "is_private_ip", scope = FunctionTemplate.FunctionScope.SIMPLE, nulls = FunctionTemplate.NullHandling.NULL_IF_NULL)
+  public static class IsPrivateIP implements DrillSimpleFunc {
+
+    @Param
+    VarCharHolder inputTextA;
+
+    @Output
+    BitHolder out;
+
+    @Inject
+    DrillBuf buffer;
+
+
+    public void setup() {
+    }
+
+
+    public void eval() {
+      String ipString = org.apache.drill.exec.expr.fn.impl.StringFunctionHelpers.toStringFromUTF8(inputTextA.start, inputTextA.end, inputTextA.buffer);
+
+      String[] ipAddressInArray = ipString.split("\\.");
+
+      int result = 0;
+
+      int[] octets = new int[3];
+
+      for (int i = 0; i < 3; i++) {
+        octets[i] = Integer.parseInt(ipAddressInArray[i]);
+        if (octets[i] > 255 || octets[i] < 0) {
```

Comment:
Why do we need this `if`?  It seems it can be removed since the subsequent if will take precedence.

Reply:
I put that in there to detect invalid IP addresses.  I guess we can remove it.